pkg/api/configmanagement/v1/types.go

```diff
@@ -528,7 +528,23 @@ type ErrorResource struct {
 	// ResourceGVK is the GroupVersionKind of the affected K8S resource. This field may be empty for
 	// errors that are not associated with a specific resource.
 	// +optional
-	ResourceGVK schema.GroupVersionKind `json:"resourceGVK"`
+	ResourceGVK GroupVersionKind `json:"resourceGVK"`
```

Comment:
I think we may need to convert this to a pointer and add omitempty so it's blank if not set. Otherwise the optional does nothing.

Even if we just add omitempty, it doesn't work on structs that are not pointers, due to a yaml formatter quirk (long standing bug).

Reply:
If we do make this optional, we need to make sure the frontend parses it correctly when missing.

I don't think we have nay other clients that read the status on the ConfigManagement object, unless nomos cli does.

Reply:
Nomos CLI doesn't parse ConfigManagement status. It only parses RSync status.

Reply:
Checking the usage of `ErrorResource`, it is not even used on the ConfigManagement object, but only on legacy mono-repo CRDs: ClusterConfig, NamespaceConfig, Repo, Sync, and etc.

Nomos CLI does parse the Repo's status field, but not `ErrorResource` field specifically. 

It should be safe to convert the field to a pointer, but do we have to make the change for monorepo code?

Reply:
> It should be safe to convert the field to a pointer, but do we have to make the change for monorepo code?

No, given that's deprecated

Reply:
Alright cool. I'm fine with this being a known bug in a deprecated API.